pkg/operators/ebpf/ebpf.go

```diff
@@ -496,6 +501,22 @@ func (i *ebpfInstance) Start(gadgetCtx operators.GadgetContext) error {
 	mapReplacements := make(map[string]*ebpf.Map)
 	constReplacements := make(map[string]any)
 
+	// create map for kernel stack
+	if _, ok := i.collectionSpec.Maps[KernelStackMapName]; ok {
+		stackIdMapSpec := ebpf.MapSpec{
+			Name:       KernelStackMapName,
+			Type:       ebpf.StackTrace,
+			KeySize:    4,
+			ValueSize:  8 * PerfMaxStackDepth,
+			MaxEntries: 10000,
```

Comment:
PerfMaxStackDepth and MAX_ENTRIES should be kept in sync between the Go source and the header file.

Add a comment `// Keep in sync with ...`.

I am also concerned with the growing API surface between ig and the gadget. Since third-party gadgets and ig are not to be released in lock-steps, we could have a gadget compiled with `stack_map.h` from an older version of ig, and then run it with a newer version of ig.

It might not matter for a map of type StackTrace, but for other kind of maps, this can cause problems. So the code pattern makes me uneasy.

I think we can accept it for now, but I am hoping we can use ebpf extensions later on.

cc @mauriciovasquezbernal

Reply:
I agree, since I have no experience in ebpf extensions, I will try it with USDT arguments first.